lib/pkgcloud/azure-v2/compute/client/servers.js

```diff
@@ -143,71 +149,110 @@ function createServer(options, callback) {
+      result.properties.outputs.hostname &&
+      result.properties.outputs.hostname.value &&
+      result.properties.outputs.location &&
+      result.properties.outputs.location.value &&
+      result.properties.outputs.vmID &&
+      result.properties.outputs.vmID.value) {
+      hostname = result.properties.outputs.hostname.value;
+      location = result.properties.outputs.location.value;
+      vmID = result.properties.outputs.vmID.value;
+    } else {
+      return callback(new Error('Result was not in the correct format: ' + JSON.stringify(result || {})));
     }
 
-    // If this is a windows machine, add an extension that enables remote connection via WinRM
     if (options.osType === 'Windows') {
-      template.resources[vmIndex].resources = [{
-        'type': 'Microsoft.Compute/virtualMachines/extensions',
-        'name': '[concat(variables(\'vmName\'),\'/WinRMCustomScriptExtension\')]',
-        'apiVersion': constants.DEFAULT_API_VERSION,
-        'location': '[resourceGroup().location]',
-        'dependsOn': [
-          '[concat(\'Microsoft.Compute/virtualMachines/\', variables(\'vmName\'))]'
-        ],
-        'properties': {
-          'publisher': 'Microsoft.Compute',
-          'type': 'CustomScriptExtension',
-          'typeHandlerVersion': '1.4',
-          'settings': {
-            'fileUris': [
-              'https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/201-vm-winrm-windows/ConfigureWinRM.ps1',
-              'https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/201-vm-winrm-windows/makecert.exe',
-              'https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/201-vm-winrm-windows/winrmconf.cmd'
-            ],
-            'commandToExecute': '[concat(\'powershell -ExecutionPolicy Unrestricted -file ConfigureWinRM.ps1 \',variables(\'hostDNSNameScriptArgument\'))]'
+      var hostnameSuffix = hostname.substring(hostname.indexOf('.') + 1);
+      var extensionId = vmID + "/extensions/WinRMCustomScriptExtension";
+      var resourceClient = new resourceManagement.ResourceManagementClient(self.azure.credentials, self.config.subscriptionId);
+      return resourceClient.resources.createOrUpdateById(
+        extensionId,
+        constants.DEFAULT_API_VERSION, {
+          "location": location,
+          "properties": {
+            "publisher": "Microsoft.Compute",
+            "typeHandlerVersion": "1.8",
+            "type": "CustomScriptExtension",
+            "settings": {
+              "fileUris": ["https://pluginsstorage.blob.core.windows.net/agentscripts/ssh.ps1"],
+              "commandToExecute": `powershell -File ssh.ps1 ${hostname}\\${options.username} ${options.password}`
+            },
+            "protectedSettings": {
+              "storageAccountName": 'pluginsstorage',
+              "storageAccountKey": 'bHabDjY34dXwITjXEasmQxI84QinJqiBZHiU+Vc1dqLNSKQxvFrZbVsfDshPriIB+XIaFVaQ2R3ua1YMDYYfHw=='
```

Comment:
Not sure what this is for, but you can't add constant personal information like:
https://pluginsstorage.blob.core.windows.net/agentscripts/ssh.ps1
pluginsstorage
bHabDjY34dXwITjXEasmQxI84QinJqiBZHiU+Vc1dqLNSKQxvFrZbVsfDshPriIB+XIaFVaQ2R3ua1YMDYYfHw==

[Same for Linux]